### DISCUSSION TITLE
How to make data available on all partials

### DISCUSSION THREAD

**Discussion by newbieexplorer212** [eta-dev/eta#83] [Q&A]
Hi, thank you for this amazing template engine.

My data is stored in json file

`eta.renderFile(file, JSON.parse(fs.readFileSync('src/eta/data.json')))`

How to make data available on all partials, so I don't have to pass data every time I `includeFile`

I can not and do not want to run the script below:
`<%~ includeFile('include/_sidebar', JSON.parse(fs.readFileSync('src/eta/data.json'))) %>`

**Comment by newbieexplorer212** [comment]
Solved using:
`<%~ includeFile('include/_sidebar', it) %>`

Since this template engine is from squirrelly, i found the solution from: https://squirrelly.js.org/docs/syntax/partials-layouts#file-partials

**Reply by urostripunovic** [reply]
How did you include a partial file? I'm trying to render a partial using this following code: 
```html
<html>
    <head>
        <meta  name="viewport" content="width=device-width, initial-scale=1.0" charset="UTF-8" />
        <title>Squirrelly middleware test</title>
    </head>
    <body>
        <h1>Test</h1>
        {{ it.title }}
        {{@includeFile("static/views/include.html")/}}
</html>
``` 
But I get the following error: `Error [Squirrelly Error]: Could not find the include file "static/views/include.html"`

Edit: I managed to solve it, you need to provide a path to where you've placed your partials when rendering the template. Squirrelly doesn't escape your path string so the error message is a bit missleading. Code snippet below shows how I solved it:
```js
const viewPath = resolve(root).replace(/\\/g, '\\');
const output = Sqrl.render(content, data, { views: [viewPath] });
```

**Comment by mylastore** [comment]
The question is why is this <%~ includeFile('include/_sidebar', it) %> not in the how to use the library, especially for Deno?
